src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx

```diff
@@ -0,0 +1,72 @@
+import { PayPalButtons } from '@paypal/react-paypal-js';
+import { setCurrentUserSubscription } from '~core/api/subscription';
+import { configRepo } from '~core/config';
+import { i18n } from '~core/localization';
+
+type PayPalButtonsGroupProps = {
+  billingPlanId: string;
+  activeBillingPlanId?: string | null;
+  activeSubscriptionId?: string | null;
+  onSubscriptionApproved?: (planId: string, subscriptionID?: string | null) => void;
+};
+
+export function PayPalButtonsGroup({
+  billingPlanId,
+  activeBillingPlanId,
+  activeSubscriptionId,
+  onSubscriptionApproved,
+}: PayPalButtonsGroupProps) {
+  return (
+    <PayPalButtons
+      disabled={!!activeBillingPlanId}
+      style={{
+        label: 'subscribe',
+        color: 'blue',
+        height: 38,
+        shape: 'rect',
+      }}
+      forceReRender={[activeSubscriptionId, activeBillingPlanId, billingPlanId]}
+      onInit={(data, actions) => {}}
+      onError={(err) => console.error('error from PayPal SDK:', err.toString())}
+      onApprove={(data, actions) => {
```

Comment:
Can we use async/await within the callbacks?

Reply:
@dqunbp depends on the callback, some of them must return Promise (e.g. `onApprove` and `createSubscription`)

Reply:
fixed